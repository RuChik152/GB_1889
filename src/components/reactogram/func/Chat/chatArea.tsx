import React, { FC } from 'react';
import style from './chat.module.scss';
import { Msg } from './Msg/msg';

type Msge = {
  msg: string;
  author: string;
  time?: string;
  id?: string;
};

interface ChatProps {
  msg: Msge[];
}

export const ChatArea: FC<ChatProps> = ({ msg }) => {
  const reversArr = () => {
    const arr = [...msg];
    return arr.reverse();
  };

  return (
    <div className={style.chat} data-testid="chat-test">
      <div className="chat__area">
        {reversArr().map((item) => (
          <Msg
            author={item.author}
            msg={item.msg}
            time={item.time}
            key={item.id}
          />
        ))}
      </div>
    </div>
  );
};
