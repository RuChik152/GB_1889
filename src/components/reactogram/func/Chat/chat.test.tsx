import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ChatArea } from './chatArea';



const msg = [
  {
    id: '11111',
    author: 'admin',
    msg: 'Тестовое сообщение!',
    time: '10:10:10',
  },
  {
    id: '22222',
    author: 'admin2',
    msg: 'Тестовое сообщение!2',
    time: '10:10:10',
  },
];

const createArr = () => {
  const count = 10000;
  const arr = [];
  for (let i = 0; i < count; i++) {
    arr.push({
      id: `00${i}`,
      author: `user_name_id_${i}`,
      msg: `Текстовое сообщение номер №${i}`,
      time: `${i} : ${i} : ${i}`,
    });
  }
  return arr;
};

describe('Chat', () => {
  it('render component', () => {
    render(<ChatArea msg={msg} />);
  });

  it('Snapshot test', () => {
    const { asFragment } = render(<ChatArea msg={msg} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Send a large array of messages', () => {
    render(<ChatArea msg={createArr()} />);
  });
});
