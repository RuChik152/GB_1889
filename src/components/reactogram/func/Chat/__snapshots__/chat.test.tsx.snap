// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="chat"
    data-testid="chat-test"
  >
    <div
      class="chat__area"
    >
      <div
        class="msg__item"
      >
        <span
          class="item__author"
          data-testid="test-author"
        >
          admin2
        </span>
        <p
          class="item__text"
          data-testid="test-text"
        >
          Тестовое сообщение!2
        </p>
        <p
          class="item__time"
          data-testid="test-time"
        >
          Время: 10:10:10
        </p>
      </div>
      <div
        class="msg__item"
      >
        <span
          class="item__author"
          data-testid="test-author"
        >
          admin
        </span>
        <p
          class="item__text"
          data-testid="test-text"
        >
          Тестовое сообщение!
        </p>
        <p
          class="item__time"
          data-testid="test-time"
        >
          Время: 10:10:10
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
